apps/omg_watcher_rpc/lib/web/validators/transaction_constraints.ex

```diff
@@ -25,17 +25,17 @@ defmodule OMG.WatcherRPC.Web.Validator.Constraints do
   @spec parse(%{binary() => any()}) :: {:ok, Keyword.t()} | {:error, any()}
   def parse(params) do
     constraints = [
-      address: [:address, :optional],
-      blknum: [:pos_integer, :optional],
-      metadata: [:hash, :optional],
-      limit: [:pos_integer, :optional],
-      page: [:pos_integer, :optional]
+      {"address", [:address, :optional]},
```

Comment:
Why?

Reply:
Originated from https://github.com/omisego/elixir-omg/pull/1192#discussion_r358220579 but still a bit off what @InoMurko suggested though because I still convert to atom after `expect/3`.

My rationale here is that the keys in this list is applied to request data, which are still strings, so 1) it's a more direct comparison against inputs, 2) it's closer to the actual boundary that this list is working on input data, and only after it passes `expect/3` that it gets `String.to_existing_atom/1` 3) a bit more performant but probably negligible.

Reply:
There were some discussions about using Ecto.Changeset as validators as well which could fit here. I'm going to merge now but very happy to continue this discussion!